src/capi/impl/capi_sframe.cpp

```diff
@@ -229,20 +232,341 @@ EXPORT void tc_sframe_random_split(const tc_sframe* sf, double fraction,
+  ERROR_HANDLE_START();
+
+  return src->value.contains_column(col_name);
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+
+EXPORT tc_sframe* tc_sframe_sample(
+    const tc_sframe* src, double fraction, uint64_t seed, tc_error** error){
+  ERROR_HANDLE_START();
+
+  return new_tc_sframe(src->value.sample(fraction, seed));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT void tc_sframe_replace_add_column(
+    tc_sframe* sf, const char* name, const tc_sarray* new_column, tc_error** error){
+  ERROR_HANDLE_START();
+
+  sf->value.replace_add_column(new_column->value, name);
+
+  ERROR_HANDLE_END(error);
+}
+
+EXPORT void tc_sframe_add_constant_column(
+    tc_sframe* sf, const char* column_name, const tc_flexible_type* value, tc_error** error){
+  ERROR_HANDLE_START();
+
+  sf->value.add_column(value->value, column_name);
+
+  ERROR_HANDLE_END(error);
+}
+
+EXPORT void tc_sframe_add_columns(
+    tc_sframe* sf, const tc_sframe* other, tc_error** error){
+  ERROR_HANDLE_START();
+
+  sf->value.add_columns(other->value);
+
+  ERROR_HANDLE_END(error);
+}
+
+EXPORT tc_sframe* tc_sframe_topk(
+    const tc_sframe* src, const char* column_name, uint64_t k, bool reverse, tc_error** error){
+  ERROR_HANDLE_START();
+
+  return new_tc_sframe(src->value.topk(column_name, k, reverse));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT void tc_sframe_swap_columns(tc_sframe* sf, const char* column_1, const char* column_2, tc_error** error){
+  ERROR_HANDLE_START();
+
+  sf->value.swap_columns(column_1, column_2);
+
+  ERROR_HANDLE_END(error);
+}
+
+EXPORT void tc_sframe_rename_column(
+    tc_sframe* sf, const char* old_name, const char* new_name, tc_error** error){
+  ERROR_HANDLE_START();
+
+  std::map<std::string, std::string> m;
+  m.insert(std::pair<std::string, std::string>(old_name, new_name));
+
+  sf->value.rename(m);
+
+  ERROR_HANDLE_END(error);
+}
+
+EXPORT void tc_sframe_rename_columns(
+    tc_sframe* sf, const tc_flex_dict* name_mapping, tc_error** error){
+  ERROR_HANDLE_START();
+
+  // TODO: write method to convert tc_flex_dict into map<str, str>
+  //sf->value.rename(name_mapping->value);
+
+  ERROR_HANDLE_END(error);
+}
+
+EXPORT tc_sframe* tc_sframe_filter_by(const tc_sframe* sf, const tc_sarray* values, const char* column_name, bool exclude, tc_error** error){
+  ERROR_HANDLE_START();
+
+  return new_tc_sframe(sf->value.filter_by(values->value, column_name, exclude));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+
+EXPORT tc_sframe* tc_sframe_pack_columns_vector(
+    const tc_sframe* sf, const tc_flex_list* columns, const char* column_name, tc_ft_type_enum type, tc_flexible_type* value, tc_error** error){
+  ERROR_HANDLE_START();
+
+  std::vector<std::string> column_transform;
+
+  for (const turi::flexible_type& i : columns->value){
+    if(i.get_type() != turi::flex_type_enum::STRING){
+      set_error(error, "columns is not of type str");
+      return NULL;
+    }
+    column_transform.push_back(i.get<turi::flex_string>());
```

Comment:
Instead of doing this in a loop like this, you can just init column_transform with

`std::vector<std::string> column_transform(columns->value.begin(), columns->value.end());`  

The rest of the casting is done automatically.